src/tests/eventpipe/GCEvents.cs

```diff
@@ -0,0 +1,255 @@
+                        Logger.logger.Log("GCRestartEEStopEvents: " + GCRestartEEStopEvents);
+                        bool GCRestartEEStartStopResult = GCRestartEEStartEvents >= 50 && GCRestartEEStopEvents >= 50;
+                        Logger.logger.Log("GCRestartEEStartStopResult check: " + GCRestartEEStartStopResult);
+
+                        Logger.logger.Log("GCSuspendEEEvents: " + GCSuspendEEEvents);
+                        Logger.logger.Log("GCSuspendEEEndEvents: " + GCSuspendEEEndEvents);
+                        bool GCSuspendEEStartStopResult = GCSuspendEEEvents >= 50 && GCSuspendEEEndEvents >= 50;
+                        Logger.logger.Log("GCSuspendEEStartStopResult check: " + GCSuspendEEStartStopResult);
+
+                        Logger.logger.Log("GCHeapStatsEvents: " + GCHeapStatsEvents);
+                        bool GCHeapStatsEventsResult = GCHeapStatsEvents >= 50 && GCHeapStatsEvents >= 50;
+                        Logger.logger.Log("GCHeapStatsEventsResult check: " + GCHeapStatsEventsResult);
+
+                        return GCStartStopResult && GCRestartEEStartStopResult && GCSuspendEEStartStopResult && GCHeapStatsEventsResult ? 100 : -1;
+                    };
+                };
+
+                var config = new SessionConfiguration(circularBufferSizeMB: (uint)Math.Pow(2, 10), format: EventPipeSerializationFormat.NetTrace,  providers: providers);
+
+                var ret = IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, config, _DoesTraceContainEvents);
+                Assert.Equal(100, ret);
+            }, output);
+        }
+
+        [Fact]
+        public async void GCWaitForPendingFinalizers_ProducesEvents()
+        {
+            await RemoteTestExecutorHelper.RunTestCaseAsync(() => 
+            {
+                Dictionary<string, ExpectedEventCount> _expectedEventCounts = new Dictionary<string, ExpectedEventCount>()
+                {
+                    { "Microsoft-Windows-DotNETRuntime", -1 },
+                    { "Microsoft-Windows-DotNETRuntimeRundown", -1 },
+                    { "Microsoft-DotNETCore-SampleProfiler", -1 }
+                };
+                
+                var providers = new List<Provider>()
+                {
+                    new Provider("Microsoft-DotNETCore-SampleProfiler"),
+                    //GCKeyword (0x1): 0b1
+                    new Provider("Microsoft-Windows-DotNETRuntime", 0b1, EventLevel.Informational)
+                };
+
+                Action _eventGeneratingAction = () => 
+                {
+                    for (int i = 0; i < 50; i++)
+                    {
+                        if (i % 10 == 0)
+                            Logger.logger.Log($"Called GC.Collect() {i} times...");
+                        TestClass testClass = new TestClass();
+                        testClass = null;
+                        GC.WaitForPendingFinalizers();
```

Comment:
You're not calling `GC.Collect()` here. `GC.WaitForPendingFinalizers()` is essentially a no-op unless the GC triggers on its own because gen 0 heap is full. Either add `GC.Collect()` or remove this line.

Reply:
Fixed by adding GC.Collect()